zookeeper/src/main/java/com/linecorp/armeria/client/endpoint/zookeeper/DefaultZkNodeValueConverter.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright 2016 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.client.endpoint.zookeeper;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import com.google.common.base.Charsets;
+
+import com.linecorp.armeria.client.Endpoint;
+
+/**
+ * A default zNode value to endpoint list converter , assuming zNode value is a CSV string
+ * like {@code "localhost:8001:5 , localhost:8002 , 192.168.1.2:80:3"} , whose each segment consists of
+ * {@code "host:portNumber:weight"}
+ * <h3>Note:</h3>
+ * <ul>
+ *   <li><b>do not</b> include schema name</li>
+ *   <li>you can omit port number and weight , so the default value will be 0 for port number
+ *   and 1000 for weight </li>
+ *   <li>must provide the port number if you specified a weight</li>
+ *  </ul>
+ */
+public class DefaultZkNodeValueConverter implements ZkNodeValueConverter {
+
+    private final Pattern segmentsPattern = Pattern.compile("[^,\\s][^\\,]*[^,\\s]*");
```

Comment:
- Missing `static`
- I'm not sure if this regular expression is correct. Why don't you just do this:

```java
private static final Pattern SEGMENT_DELIMITER = Pattern.compile("\\s*,\\s*");
...
String valueString = new String(data, StandardCharsets.UTF_8);
for (String segment : SEGMENT_DELIMITER.split(valueString)) {
    String[] tokens = segment.split(":");
    switch (tokens.length) { ...
```